### PR TITLE
refactored google authentication by adding react-google-login library

### PR DESCRIPTION
**Pull request by Haytham1337**
Replace Goggle api authentication with react-google-login.

### CODE REVIEW

File: src/pages/login/login.js [hunk 1]
```diff
@@ -64,21 +65,11 @@ const Login = () => {
...
-    });
-  }, []);
-
-  const singIn = () => {
-    const googleAuth = window.gapi.auth2.getAuthInstance();
-    googleAuth.signIn().then((googleUser) => {
-      const idToken = googleUser.getAuthResponse().id_token;
-      dispatch(loginByGoogle({ idToken }));
-    });
+  const responseGoogleSuccess = (response) => {
```
Comment (moran711) [inline src/pages/login/login.js:68]: You can destructure response like `{tokenId}`

File: src/pages/login/login.js [hunk 2]
```diff
@@ -165,14 +156,24 @@ const Login = () => {
...
+                            onClick={renderProps.onClick}
+                            disabled={renderProps.disabled}
+                            className={styles.googleBtn}
+                            fullWidth
+                          >
+                            <span className={styles.googleLogo} />
+                            Google
+                          </Button>
+                        )}
+                        buttonText='Login with Google'
```
Comment (moran711) [inline src/pages/login/login.js:172]: Move this magic string to configs

File: src/pages/login/login.js [hunk 3]
```diff
@@ -165,14 +155,23 @@ const Login = () => {
...
+                            className={styles.googleBtn}
+                            fullWidth
+                          >
+                            <span className={styles.googleLogo} />
+                            Google
+                          </Button>
+                        )}
+                        onSuccess={responseGoogleSuccess}
+                        onFailure={responseGoogleFailure}
+                        cookiePolicy='single_host_origin'
```
Comment (kapoliub) [inline src/pages/login/login.js:173]: move this string to config file

**Review by kapoliub — COMMENTED** [review]
Screen Shot 2021-02-03 at 14 43 10
button is disabled

**Review by moran711 — APPROVED** [review]

**Review by kapoliub — APPROVED** [review]

**Review by NikitaDenysenko — APPROVED** [review]

**Pull request merged**